lib/process-services-cloud/src/lib/group/components/group-cloud.component.ts

```diff
@@ -214,10 +217,14 @@ export class GroupCloudComponent implements OnInit, OnChanges {
             this.preSelectGroups.forEach((group: GroupModel) => {
                 this.selectedGroups.push(group);
             });
+            const groups = this.removeDuplicatedGroups(this.selectedGroups);
+            this.selectedGroups = [...groups];
             this.selectedGroupsSubject.next(this.selectedGroups);
         } else {
-            this.searchGroupsControl.setValue(this.preSelectGroups[0]);
-            this.onSelect(this.preSelectGroups[0]);
+            setTimeout(() => {
```

Comment:
This method is called on the ngOnChanges so can potentially spawn many timers working in parallel. I suggest cleaning the previous timer instead, see for the example: https://www.w3schools.com/jsref/met_win_cleartimeout.asp